### PR TITLE
added section on adding csv data to the database

### PR DESCRIPTION
**Pull request by avermeulen**
@SteveBarnett I moved this into the database repo - I think this is more general than Nelisa itself although one part of it is specific to her. Thoughts please? @Oyamasiphula @Onwa2014

### CODE REVIEW

File: steps/add_csv_data_to_db.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+layout: default
+---
+
+# Add data to your database
+
+Populating a relational database from CSV files can be tricky due to the relationships between tables. The relationships are enforced by foreign keys constraints on each table. The foreign key constraints enforce the correct relationships between tables, that is that the foreign key columns contains values that exists in the corresponding column in the referenced table. As a result the order in which rows are added to the tables in a relational database is very important.
+
+Say for example you have a football team database with tables for `Positions`, `Players` and `Teams`. Each player have a position and and each team have multiple players, the database use a `TeamPlayers` table, to link players to teams. The database have foreign key constraints to ensure that `Players` can only use positions that exists in the `Positions` table and that the `TeamPlayers` table can only cont
```
Comment (SteveBarnett) [inline steps/add_csv_data_to_db.md]: @avermeulen Unfinished sentence? 

> can only cont

File: steps/add_csv_data_to_db.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+To populate this database from a CSV file you will need to:
+
+* populate the `Positions` table;
+* populate the `Players` table using the foreign key from the Position player;
+* populate the `Teams` table;
+* populate the `TeamPlayers` table using foreign keys from the `Teams` and the `Players` table.
+
+## Database for Nelisa
+
+Once you have a proper database schema in place with foreign keys between the tables, you you will need to add the data in this order:
```
Comment (SteveBarnett) [inline steps/add_csv_data_to_db.md]: Worth explicitly stating what a Schema is, since we haven't yet?

File: steps/add_csv_data_to_db.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+* Add the Categories to the database;
```
Comment (SteveBarnett) [inline steps/add_csv_data_to_db.md]: Should we change this to match the example above? "Create and populate the Categories table" etc.

File: steps/add_csv_data_to_db.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+* add Products to the database with foreign keys that reference the Categories table;
+* add Sales & Purchase records to the database, with foreign keys that references entries in the Products table.
+
+## Adding Categories
+
+This one is fairly straightforward, loop through your list of Categories and then to a list of lists and use a [bulk insert](/steps/mysql_bulk_insert_using_nodejs.html) add them to the database. The order of the list for each row should match your `sql` insert statement.
```
Comment (SteveBarnett) [inline steps/add_csv_data_to_db.md]: This doesn't quite make sense. "loop through your list of Categories and then to a list of lists and use a". Do you mean: "loop through your list of Categories and _create_ a list of lists and use a"?

**Comment by SteveBarnett** [comment]
@avermeulen I've pushed some language fixes and left a comment or two for your perusal.

File: steps/add_csv_data_to_db.md (hunk 1 above)
Comment (avermeulen) [inline steps/add_csv_data_to_db.md]: I completed the sentence @SteveBarnett

File: steps/add_csv_data_to_db.md (hunk 4 above)
Comment (avermeulen) [inline steps/add_csv_data_to_db.md]: @SteveBarnett good spot! It makes more sense now...

File: steps/add_csv_data_to_db.md (hunk 2 above)
Comment (avermeulen) [inline steps/add_csv_data_to_db.md]: @SteveBarnett  - I always cover this: http://database.projectcodex.co/steps/data_modelling.html
As part of a brief Database workshop I will extend the content to mention schema - or we can replace schema with Data Model.

**Comment by avermeulen** [comment]
@SteveBarnett I made a bunch of changes on the back of your comments, can you please review it.

**Comment by Oyamasiphula** [comment]
Ok thats going to be my next thing to do just after reviewing The
deployment process for Onwa. Is that going ot OK?

On Mon, May 30, 2016 at 2:36 AM, Andre Vermeulen notifications@github.com
wrote:

> @SteveBarnett https://github.com/SteveBarnett I made a bunch of changes
> on the back of your comments, can you please review it.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/codex-academy/database-intro/pull/4#issuecomment-222453793,
> or mute the thread
> https://github.com/notifications/unsubscribe/AIXx6LLUkdZOjSObPP1pLiTvDj-Z977Fks5qGq-rgaJpZM4IomQ5
> .

## 

O.siphula

**Pull request merged**